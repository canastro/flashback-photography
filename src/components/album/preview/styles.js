import {scale, rhythm} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'block',
        backgroundColor: 'lightgray',
        flex: '1 0 0%',
        marginRight: rhythm(1 / 8),
        width: '100%',
        maxWidth: 290.1,
        position: 'relative',
        [presets.Tablet]: {
            marginRight: rhythm(1)
        },
        ':last-child': {
            marginRight: 0
        },
        color: 'inherit',
        textTransform: 'uppercase',
        textDecoration: 'none'
    },
    wrapper: {
        flexDirection: 'column',
        flexShrink: 0,
        position: 'relative',
        overflow: 'hidden',
        outline: 0
    },
    image: {
        margin: 0,
        height: '100%',
        width: '100%',
        verticalAlign: 'baseline',
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        outline: 'none'
    },
    overlay: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'rgba(0,0,0,0.3)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        color: 'white',
        ...scale(2 / 5)
    },
    icon: {
        fontSize: '90%',
        marginRight: rhythm(1 / 4)
    },
    title: {
        position: 'absolute',
        bottom: 0,
        height: 40,
        backgroundColor: 'white',
        width: '100%',
        opacity: 0.8,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
    }
};
