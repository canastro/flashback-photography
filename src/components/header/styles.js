import {rhythm, scale} from '../../utils/typography';

export const burgerStyles = {
    bmBurgerButton: {
        position: 'absolute',
        width: '36px',
        height: '30px',
        left: 'initial',
        top: 20,
        right: 36
    },
    bmBurgerBars: {
        background: '#373a47'
    },
    bmCrossButton: {
        height: '24px',
        width: '24px'
    },
    bmCross: {
        background: '#bdc3c7'
    },
    bmMenuWrap: {
        top: 0
    },
    bmMenu: {
        background: '#373a47',
        padding: '2.5em 1.5em 0',
        fontSize: '1.15em'
    },
    bmMorphShape: {
        fill: '#373a47'
    },
    bmItemList: {
        color: '#b8b7ad',
        padding: '0.8em',
        display: 'flex',
        flexDirection: 'column'
    },
    bmOverlay: {
        background: 'rgba(0, 0, 0, 0.3)',
        left: 0,
        top: 0
    }
};

export default {
    root: {
        background: 'white',
        borderBottom: '1px solid rgba(0,0,0,0.08)',
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        zIndex: 1
    },
    wrapper: {
        maxWidth: 960,
        margin: '0 auto',
        overflow: 'hidden',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    titleLink: {
        padding: rhythm(3 / 4),
        display: 'inline-block',
        float: 'left',
        textDecoration: 'none'
    },
    title: {
        ...scale(4 / 5),
        lineHeight: 1,
        margin: 0,
        overflow: 'hidden'
    },
    titleSpan: {
        paddingLeft: `calc(${rhythm(1)} - 1px)`,
        borderLeft: '1px solid rgba(0,0,0,0.3)',
        lineHeight: 1,
        marginLeft: rhythm(1)
    },
    icon: {
        top: -4,
        display: 'inline-block',
        position: 'relative'
    },
    link: {
        color: 'inherit',
        display: 'inline-block',
        float: 'right',
        lineHeight: '35px',
        textDecoration: 'none',
        marginLeft: 20,
        '&:hover': {
            color: '#000'
        }
    }
};
