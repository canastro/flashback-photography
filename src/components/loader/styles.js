export default {
    root: {
        position: 'fixed',
        height: '100%',
        width: '100%',
        top: 0,
        left: 0,
        backgroundColor: 'black',
        opacity: 0.5,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        zIndex: 10
    },
    indicator: {
        display: 'inline-block',
        color: '#fff',
        width: 100
    },
    svgIndeterminate: {
        animation: 'circular-rotate 1.4s linear infinite'
    },
    circle: {
        stroke: 'currentColor',
        strokeLinecap: 'round',
        animation: 'circular-dash 1.4s ease-in-out infinite',
        // Some default value that looks fine waiting for the animation to kicks in.
        strokeDasharray: '80,200',
        strokeDashoffset: 0
    },
    '@keyframes circular-rotate': {
        '100%': {
            transform: 'rotate(360deg)'
        }
    },
    '@keyframes circular-dash': {
        '0%': {
            strokeDasharray: '1,200',
            strokeDashoffset: 0
        },
        '50%': {
            strokeDasharray: '100,200',
            strokeDashoffset: -15
        },
        '100%': {
            strokeDasharray: '100,200',
            strokeDashoffset: -120
        }
    }
};
