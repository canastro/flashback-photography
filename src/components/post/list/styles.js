import {rhythm, scale} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'flex',
        alignItems: 'stretch',
        flexShrink: 0,
        flexDirection: 'column'
    },
    postsChunk: {
        display: 'flex',
        alignItems: 'stretch',
        flexShrink: 0,
        flexDirection: 'row',
        marginBottom: rhythm(1 / 8),
        [presets.Tablet]: {
            marginBottom: rhythm(1)
        }
    },
    showMore: {
        ...scale(-0.5),
        border: '1px solid black',
        boxShadow: 0,
        background: 'none',
        color: 'black',
        cursor: 'pointer',
        margin: '0 auto',
        padding: rhythm(1 / 2),
        width: `calc(100vw - ${rhythm(1)})`,
        marginLeft: rhythm(0.5),
        marginRight: rhythm(0.5),
        marginBottom: rhythm(0.5),
        marginTop: rhythm(0.5),
        [presets.Tablet]: {
            margin: '0 auto',
            marginBottom: rhythm(1.5),
            marginTop: rhythm(1.5),
            height: rhythm(2),
            width: rhythm(5),
            textAlign: 'center'
        },
        '&:hover': {
            backgroundColor: '#ccc'
        }
    }
};
