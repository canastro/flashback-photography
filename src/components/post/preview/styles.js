import {scale, rhythm} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'block',
        backgroundColor: 'black',
        flex: '1 0 0%',
        marginRight: rhythm(1 / 8),
        width: '100%',
        maxWidth: 290.1,
        position: 'relative',
        [presets.Tablet]: {
            marginRight: rhythm(1)
        },
        ':last-child': {
            marginRight: 0
        }
    },
    wrapper: {
        width: '100%',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center'
    },
    image: {
        maxHeight: 200
    },
    overlay: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'rgba(0,0,0,0.3)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        color: 'white',
        ...scale(2 / 5)
    },
    icon: {
        fontSize: '90%',
        marginRight: rhythm(1 / 4)
    }
};
