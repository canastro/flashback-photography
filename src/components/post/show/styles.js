import presets from '../../../utils/presets';
import typography, {rhythm, scale} from '../../../utils/typography';

export default {
    root: {
        maxHeight: 500,
        backgroundColor: 'black',
        display: 'flex',
        alignItems: 'stretch',
        flexDirection: 'column',
        width: '100%',
        [presets.Tablet]: {
            flexDirection: 'row-reverse'
        }
    },
    detailsContainer: {
        ...scale(-2 / 5),
        backgroundColor: 'white',
        lineHeight: typography.options.baseLineHeight,
        height: '100%'
    },
    detailsContent: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-around',
        height: 'calc(100% - 80px)'
    },
    date: {
        color: 'rgba(0,0,0,0.4)',
        float: 'right'
    },
    tabletDetailsContainer: {
        backgroundColor: 'white',
        padding: rhythm(3 / 4),
        paddingBottom: 0,
        [presets.Tablet]: {
            width: rhythm(13),
            padding: rhythm(1)
        }
    },
    tabletDetailsWrapper: {
        display: 'none',
        [presets.Tablet]: {
            display: 'block',
            height: '100%'
        }
    },
    mobileDetailsWrapper: {
        background: 'white',
        padding: rhythm(3 / 4),
        display: 'block',
        [presets.Tablet]: {
            display: 'none'
        }
    },
    imageContainer: {
        backgroundColor: 'lightgray',
        flex: '1 0 0',
        margin: 'auto',
        width: '100%'
    },
    listContainer: {
        display: 'flex',
        flexDirection: 'column'
    },
    settingItem: {
        listStyle: 'none',
        padding: 0,
        height: 25,
        display: 'flex',
        alignItems: 'center'
    },
    icon: {
        marginRight: 10
    },
    tag: {
        backgroundColor: '#ccc',
        margin: '10px 0 0 0',
        textAlign: 'center',
        width: 100,
        color: 'inherit',
        textDecoration: 'none',
        '&:hover': {
            backgroundColor: '#696969'
        }
    }
};
