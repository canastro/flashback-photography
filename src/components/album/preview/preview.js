// @flow
import React from 'react';
import Link from 'gatsby-link';
import injectSheet from 'react-jss';
import EyeIcon from 'react-icons/lib/fa/eye';
import slug from 'slug';

import styles from './styles';

type Props = {
    classes: Object,
    album: Object
};

/**
 * Create a post preview
 * @method Preview
 * @param  {Object} props - react props
 * @returns {Node} Preview preview
 */
export class Preview extends React.Component {
    props: Props;

    /**
     * Constructor, initializes state
     * @method  constructor
     * @param   {Object} props - react props
     */
    constructor(props) {
        super(props);
        this.state = {
            hovering: false
        };
    }

    /**
     * Renders the post preview
     * @method  render
     * @returns {Node} - react node
     */
    render() {
        const {classes, album} = this.props;
        const {cover, name} = album;
        return (
            <Link
                to={`/album/${slug(name)}/`}
                className={classes.root}
                onMouseEnter={() => {
                    this.setState({hovering: true});
                }}
                onMouseLeave={() => {
                    this.setState({hovering: false});
                }}
            >
                <div className={classes.wrapper}>
                    <img
                        alt="Preview Preview"
                        src={cover.responsiveResolution.src}
                        srcSet={cover.responsiveResolution.srcSet}
                        className={classes.image}
                    />
                </div>

                {this.state.hovering && (
                    <div className={classes.overlay}>
                        <EyeIcon className={classes.icon} />
                    </div>
                )}
                <div className={classes.title}>{name}</div>
            </Link>
        );
    }
}

export default injectSheet(styles)(Preview);
