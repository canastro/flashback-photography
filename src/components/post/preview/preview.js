// @flow
import React from 'react';
import Link from 'gatsby-link';
import injectSheet from 'react-jss';
import EyeIcon from 'react-icons/lib/fa/eye';

import styles from './styles';

type Props = {
    classes: Object,
    post: Object
};

/**
 * Create a post preview
 * @method Post
 * @param  {Object} props - react props
 * @returns {Node} Post preview
 */
class Post extends React.Component {
    props: Props;

    /**
     * Constructor, initializes state
     * @method  constructor
     * @param   {Object} props - react props
     */
    constructor(props) {
        super(props);
        this.state = {
            hovering: false
        };
    }

    /**
     * Renders the post preview
     * @method  render
     * @returns {Node} - react node
     */
    render() {
        const {classes, post} = this.props;
        const {preview, id} = post;
        return (
            <Link
                to={`/post/${id}/`}
                className={classes.root}
                onMouseEnter={() => {
                    this.setState({hovering: true});
                }}
                onMouseLeave={() => {
                    this.setState({hovering: false});
                }}
            >
                <div className={classes.wrapper}>
                    <img
                        alt="Post Preview"
                        src={preview.responsiveResolution.src}
                        srcSet={preview.responsiveResolution.srcSet}
                        className={classes.image}
                    />
                </div>
                {/* overlay */}
                {this.state.hovering && (
                    <div className={classes.overlay}>
                        <EyeIcon className={classes.icon} />
                    </div>
                )}
            </Link>
        );
    }
}

export default injectSheet(styles)(Post);

export const postFragment = graphql`
    fragment Post_details on ContentfulPost {
        id
        preview: photo {
            responsiveResolution(width: 200) {
                width
                height
                src
                srcSet
            }
        }
    }
`;
