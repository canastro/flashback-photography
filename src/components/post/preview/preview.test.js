import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {Preview} from './preview';
import styles from './styles';
import {getClasses} from '../../../utils/tests';


const classes = getClasses(styles);

const post = {
    id: 999,
    preview: {
        sizes: {
            test: 'TEST-SIZES'
        }
    }
};

it('should render correctly', () => {
    const output = shallow(<Preview classes={classes} post={post} />);
    expect(shallowToJson(output)).toMatchSnapshot();
});

describe('when mouse enters link', () => {
    let output;

    beforeAll(() => {
        output = shallow(<Preview classes={classes} post={post} />);
        const link = output.find('GatsbyLink');

        expect(output.state().hovering).toEqual(false);
        link.prop('onMouseEnter')();
        output.update();
    });

    it('should update state', () => {
        expect(output.state().hovering).toEqual(true);
    });

    it('should render eye icon', () => {
        expect(output.find('FaEye')).toHaveLength(1);
    });

    it('should render correctly', () => {
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});

describe('when mouse leaves link', () => {
    let output;

    beforeAll(() => {
        output = shallow(<Preview classes={classes} post={post} />);
        output.setState({hovering: true});

        const link = output.find('GatsbyLink');

        expect(output.state().hovering).toEqual(true);
        link.prop('onMouseLeave')();
    });

    it('should update state', () => {
        expect(output.state().hovering).toEqual(false);
    });

    it('should render correctly', () => {
        output.update();
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
