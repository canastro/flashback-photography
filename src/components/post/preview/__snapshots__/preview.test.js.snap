// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<GatsbyLink
  className="root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  to="/post/999/"
>
  <div
    className="wrapper"
  >
    <Image
      alt="Post Preview"
      className="image"
      fadeIn={true}
      sizes={
        Object {
          "test": "TEST-SIZES",
        }
      }
    />
  </div>
</GatsbyLink>
`;

exports[`when mouse enters link should render correctly 1`] = `
<GatsbyLink
  className="root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  to="/post/999/"
>
  <div
    className="wrapper"
  >
    <Image
      alt="Post Preview"
      className="image"
      fadeIn={true}
      sizes={
        Object {
          "test": "TEST-SIZES",
        }
      }
    />
  </div>
  <div
    className="overlay"
  >
    <FaEye
      className="icon"
    />
  </div>
</GatsbyLink>
`;

exports[`when mouse leaves link should render correctly 1`] = `
<GatsbyLink
  className="root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  to="/post/999/"
>
  <div
    className="wrapper"
  >
    <Image
      alt="Post Preview"
      className="image"
      fadeIn={true}
      sizes={
        Object {
          "test": "TEST-SIZES",
        }
      }
    />
  </div>
</GatsbyLink>
`;
